spec/factories/service_providers.rb

```diff
@@ -15,6 +15,14 @@
       identity_protocol { 1 }
     end
 
+    trait :with_oidc_jwt do
+      identity_protocol { 0 }
+    end
+
+    trait :with_oidc_pke do
+      identity_protocol { 2 }
+    end
```

Comment:
1. I am pretty sure we can assign the enum names here for readability:
2. should that be pk_c_e?

```suggestion
    trait :with_oidc_jwt do
      identity_protocol { :openid_connect_private_key_jwt }
    end

    trait :with_oidc_pcke do
      identity_protocol { :openid_connect_pkce }
    end
```